### PR TITLE
Don't depend on X11 in the MetafileTests

### PR DESCRIPTION
**Pull request by qmfrederik**
This will allow us to re-enable most tests on Unix.

### CODE REVIEW

File: src/System.Drawing.Common/src/System/Drawing/Imaging/Metafile.cs [hunk 1]
```diff
@@ -243,8 +243,11 @@ public Metafile(IntPtr referenceHdc, Rectangle frameRect, MetafileFrameUnit fram
         /// </summary>
         public Metafile(string fileName, IntPtr referenceHdc, EmfType type, String description)
         {
-            // Called in order to emulate exception behavior from netfx related to invalid file paths.
-            Path.GetFullPath(fileName);
+            if (fileName == null)
```
Comment (mellinoe) [inline src/System.Drawing.Common/src/System/Drawing/Imaging/Metafile.cs]: Hmm, isn't this effectively what `Path.GetFullPath` will do?

The .NET Framework codebase definitely calls `Path.GetFullPath`, it's just done indirectly through `IntSecurity.DemandReadFileIO(fileName);`.

http://referencesource.microsoft.com/#System.Drawing/commonui/System/Drawing/Advanced/Metafile.cs,358

**Comment by mellinoe** [comment]
Looks good overall, but I think we should use the same `Path.GetFullPath` pattern everywhere, if we want to throw those exceptions the same way that .NET Framework does.

**Comment by mellinoe** [comment]
@qmfrederik Sorry about this -- I moved the remaining files under src/Unix into the regular directory structure with #24194 . I only moved them and didn't actually change their contents.

I assume you want to get this change in before #24151?

**Comment by qmfrederik** [comment]
@mellinoe This PR is independent from #24151. In #24151, I re-enabled some tests which were included in Mono's test suite. This PR tries to re-enable all Metafile tests on Linux. So #24151 can go first.

**Comment by qmfrederik** [comment]
@mellinoe I have a question regarding `Ctor_PathTooLongI_ThrowsPathTooLongException`. This one was tracked as part of #22723 which is now marked as closed.

I re-enabled the test but it still fails; it seems `Path.GetFullPath` doesn't throw a `ThrowsPathTooLongException`, but GDI+ still barks at the path.

Do you know of any other method we can call which would catch file names which are too long for GDI+ to handle?

**Comment by mellinoe** [comment]
> Do you know of any other method we can call which would catch file names which are too long for GDI+ to handle?

I'm not sure. Perhaps we should just hard-code it to 255 characters (or whatever the limit is?) Using `Path.GetFullPath` might have just been a roundabout way to get that when the behavior used to match.

File: src/System.Drawing.Common/src/System/Drawing/Imaging/Metafile.cs (hunk 1 above)
Comment (qmfrederik) [reply src/System.Drawing.Common/src/System/Drawing/Imaging/Metafile.cs]: Yes, but `Path.GetFullPath` will set `path` as the argument name, whereas the unit tests expect it to be `null`; see e.g. `Ctor_InvalidPathII_ThrowsArgumentException`.

I guess in this case the .NET Core behavior is an improvement and we could amend the tests to assert that ParamName is `path` on Core?

Comment (mellinoe) [reply src/System.Drawing.Common/src/System/Drawing/Imaging/Metafile.cs]: > I guess in this case the .NET Core behavior is an improvement and we could amend the tests to assert that ParamName is path on Core?
 
Yeah, we have some functions in AssertExtensions.cs that allow you to specify a different string name on core / netfx to allow for this. We use it in some other parts of the test.

**Comment by qmfrederik** [comment]
@dotnet-bot test UWP CoreCLR x64 Debug Build please

**Comment by qmfrederik** [comment]
@mellinoe I think this one is ready to go.

File: src/System.Drawing.Common/tests/Imaging/MetafileTests.cs [hunk 2]
```diff
@@ -756,6 +757,7 @@ public void Ctor_NullPathI_ThrowsArgumentNullException()
             }
         }
 
+        [ActiveIssue(20844, TestPlatforms.AnyUnix)]
```
Comment (am11) [inline src/System.Drawing.Common/tests/Imaging/MetafileTests.cs]: Is this issue opened in CoreFX repo? Seems like #20844 is a PR.

Comment (qmfrederik) [reply src/System.Drawing.Common/tests/Imaging/MetafileTests.cs]: Should be #20884, I'll fix that.

**Comment by qmfrederik** [comment]
@dotnet-bot test UWP CoreCLR x64 Debug Build please

**Pull request merged**